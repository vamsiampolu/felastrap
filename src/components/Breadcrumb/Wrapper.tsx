import { useFela } from "react-fela";
import type { CSSProperties } from "fela";
import type { ComponentPropsWithoutRef } from "react";
import type { Theme, StyleProps } from "../common";

export type WrapperProps = ComponentPropsWithoutRef<"ol"> & {
  backgroundColor: string | null;
  borderRadius: string | number | null;
};

const rule = ({
  backgroundColor = null,
  borderRadius = null,
}: StyleProps<WrapperProps>) => {
  const padding = {
    paddingLeft: 0,
    paddingRight: 0,
    paddingTop: 0,
    paddingBottom: 0,
  };

  const style: CSSProperties = {
    ...padding,
    display: "flex",
    flexWrap: "wrap",
    marginBottom: "1rem",
    fontSize: "1rem",
    listStyle: "none",
  };

  if (backgroundColor) {
    style.backgroundColor = backgroundColor;
  }

  if (borderRadius) {
    style.borderRadius = borderRadius;
  }

  return style;
};

export const Wrapper = ({
  borderRadius = null,
  backgroundColor = null,
  children,
  ...other
}: WrapperProps) => {
  const { css } = useFela<Theme, WrapperProps>({
    borderRadius,
    backgroundColor,
  });

  const className = css(rule);
  return (
    <nav data-testid="breadcrumb" aria-label="Breadcrumb">
      <ol className={className} {...other}>
        {children}
      </ol>
    </nav>
  );
};
