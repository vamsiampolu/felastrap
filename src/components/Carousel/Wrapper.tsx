import { DivProps } from "../common";
import { useFela } from "react-fela";

export const Wrapper = ({ className = "", children, ...other }: DivProps) => {
  const { css } = useFela();
  const cx = `${css({ position: "relative" })} ${className} carousel`;
  return (
    <div data-testid="carousel" className={cx} {...other}>
      {children}
    </div>
  );
};
