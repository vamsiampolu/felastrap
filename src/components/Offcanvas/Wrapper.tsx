import { CSSCustomPrimitive, CSSProperties, IStyle } from "fela";
import { ElementType, forwardRef } from "react";
import { useFela } from "react-fela";
import {
  PolymorphicComponentPropsWithoutRef,
  PolymorphicRef,
} from "../../polymorphicComponentProps";
import { StyleProps, Theme } from "../common";

export type OffcanvasPlacement = "start" | "top" | "end" | "bottom";
export type CustomWrapperProps = {
  id: string;
  titleId: string;
  width?: CSSProperties["width"];
  isOpen?: boolean;
  placement?: OffcanvasPlacement;
};

type RuleProps = Omit<CustomWrapperProps, "id" | "titleId">;

export type WrapperProps<T extends ElementType> =
  PolymorphicComponentPropsWithoutRef<T, CustomWrapperProps>;

export type RuleResult = IStyle | CSSCustomPrimitive;

const rule = (): RuleResult => ({
  position: "fixed",
  bottom: 0,
  zIndex: 1045,
  display: "flex",
  flexDirection: "column",
  maxWidth: "100%",
  color: "#212529",
  visibility: "hidden",
  backgroundColor: "#fff",
  backgroundClip: "padding-box",
  outline: 0,
  transition: "transform 0.3s ease-in-out",
});

const startRule = ({ placement, width }: StyleProps<RuleProps>) =>
  placement === "start"
    ? {
        top: 0,
        left: 0,
        width: width ? width : "400px",
        borderRight: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateX(-100%)",
      }
    : {};

const endRule = ({ placement, width }: StyleProps<RuleProps>): RuleResult =>
  placement === "end"
    ? {
        top: 0,
        right: 0,
        width: width ? width : "400px",
        borderLeft: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateX(100%)",
      }
    : {};

const topRule = ({ placement }: StyleProps<RuleProps>): RuleResult =>
  placement === "top"
    ? {
        top: 0,
        right: 0,
        left: 0,
        height: "auto",
        maxHeight: "100%",
        borderBottom: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateY(-100%)",
      }
    : {};

const bottomRule = ({ placement }: StyleProps<RuleProps>): RuleResult =>
  placement === "bottom"
    ? {
        right: 0,
        left: 0,
        height: "auto",
        maxHeight: "100%",
        borderTop: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateY(100%)",
      }
    : {};

const showStartRule = ({
  placement,
  isOpen,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "start"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateX(-100%)",
        visibility: "hidden",
      };

const showEndRule = ({
  placement,
  isOpen,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "end"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateX(100%)",
        visibility: "hidden",
      };

const showTopRule = ({
  isOpen,
  placement,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "top"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateY(-100%)",
        visibility: "hidden",
      };

const showBottomRule = ({
  isOpen,
  placement,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "bottom"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateY(100%)",
        visibility: "hidden",
      };

const showPlacementRule = ({
  isOpen,
  placement,
}: StyleProps<RuleProps>): RuleResult => {
  switch (placement) {
    case "bottom": {
      return showBottomRule({
        isOpen,
        placement,
      } as StyleProps<RuleProps>);
    }

    case "end": {
      return showEndRule({
        isOpen,
        placement,
      } as StyleProps<RuleProps>);
    }

    case "top": {
      return showTopRule({ isOpen, placement } as StyleProps<RuleProps>);
    }

    case "start": {
      return showStartRule({
        isOpen,
        placement,
      } as StyleProps<RuleProps>);
    }
  }

  return {};
};

export const WrapperComponent = <T extends ElementType = "div">(
  {
    as,
    id,
    className,
    isOpen = false,
    titleId,
    children,
    placement = "start",
    width = undefined,
    ...props
  }: WrapperProps<T>,
  ref: PolymorphicRef<T>,
) => {
  const { css } = useFela<Theme, RuleProps>({ isOpen, placement, width });
  const cx = `${css([
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    rule as any,
    startRule,
    endRule,
    topRule,
    bottomRule,
    showPlacementRule,
  ])} offcanvas offcanvas-${placement} ${className || ""}`;
  const Component = as || "div";
  return (
    <Component
      data-testid="offcanvas-wrapper"
      ref={ref}
      className={cx}
      tabIndex={-1}
      id={id}
      aria-labelledby={titleId}
      {...props}
    >
      {children}
    </Component>
  );
};

export const Wrapper = forwardRef(WrapperComponent);
Wrapper.displayName = "Wrapper";
