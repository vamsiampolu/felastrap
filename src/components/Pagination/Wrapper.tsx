import { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";

export type WrapperProps = ComponentPropsWithoutRef<"ul"> &
  Partial<{ size: number }>;

export const Wrapper = ({
  className = "",
  children,
  ...other
}: WrapperProps) => {
  const { css } = useFela();
  const style = {
    display: "flex",
    listStyle: "none",
    paddingLeft: 0,
  };
  const label = other["aria-label"] || "Page Navigation";
  const cx = `${css(style)} ${className}`;
  return (
    <nav data-testid="pagination-wrapper" aria-label={label}>
      <ul className={cx}>{children}</ul>
    </nav>
  );
};
