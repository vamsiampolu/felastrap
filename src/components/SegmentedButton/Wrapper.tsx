import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  position: "relative",
  display: "flex",
  flexWrap: "wrap",
  alignItems: "stretch",
  width: "100%",
  marginBottom: "1rem",
});

export const Wrapper = ({ className = "", children, ...props }: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} segmented-btn-wrapper ${className}`.trim();
  return (
    <div data-testid="segmented-button" className={cx} {...props}>
      {children}
    </div>
  );
};
