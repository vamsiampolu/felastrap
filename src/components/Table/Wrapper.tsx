import { ComponentPropsWithoutRef, useContext } from "react";
import { useFela } from "react-fela";
import { variantRule } from "./variant";
import { TableContext } from "./TableContext";

const rule = () => ({
  width: "100%",
  marginBottom: "1rem",
  verticalAlign: "top",
  borderColor: "#dee2e6",
});

const borderlessRule = () => ({
  ":not(:first-child)": {
    borderTopWidth: 0,
  },
});

export type WrapperProps = ComponentPropsWithoutRef<"table">;

export const Wrapper = ({
  className = "",
  children,
  ...props
}: WrapperProps) => {
  const {
    hover = false,
    borderless = false,
    variant = "default",
  } = useContext(TableContext);
  const { css } = useFela({ variant, borderless, hover });
  return (
    <table
      data-testid="table"
      className={`${css([
        rule,
        variantRule,
        borderlessRule,
      ])} bs-table ${className}`.trim()}
      {...props}
    >
      {children}
    </table>
  );
};
