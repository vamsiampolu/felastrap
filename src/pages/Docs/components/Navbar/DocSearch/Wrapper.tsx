import { ReactNode } from "react";
import { useFela } from "react-fela";

const searchButtonRule = () => ({
  position: "relative",
  "@media (min-width: 1200px)": {
    width: "200px",
    marginLeft: "-140px",
  },
  "@media (min-width: 992px)": {
    position: "absolute !important",
    top: "0.65rem",
    left: "50%",
    width: "200px",
    marginLeft: "100px",
  },
});

export const DocSearchWrapper = ({ children }: { children: ReactNode }) => {
  const { css } = useFela();
  return (
    <div className={`${css({ display: "flex" })}`}>
      <div
        className={`${css(searchButtonRule)} bd-search`}
        id="docsearch"
        data-bd-docs-version="5.3"
      >
        {children}
      </div>
    </div>
  );
};
