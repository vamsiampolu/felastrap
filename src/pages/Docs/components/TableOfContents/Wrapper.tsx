import { useFela } from "react-fela";
import { DivProps } from "../../../../components/common";

const rule = () => ({
  color: "rgba(33, 37, 41, 0.75)",
  gridArea: "toc",
  marginTop: "1rem",
  opacity: 1,
  "@media (min-width: 576px)": {
    paddingRight: "0.25rem",
    paddingLeft: "0.25rem",
  },
  "@media (min-width: 992px)": {
    position: "sticky",
    top: "5rem",
    right: "0",
    zIndex: "2",
    height: "calc(100vh - 7rem)",
    overflowY: "auto",
    marginBottom: "3rem",
    marginTop: 0,
  },
});

export const Wrapper = ({ className = "", children }: DivProps) => {
  const { css } = useFela();
  return (
    <div className={`${css(rule)} toc ${className}`.trim()}>{children}</div>
  );
};
