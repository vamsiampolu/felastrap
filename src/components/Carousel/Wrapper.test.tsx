import { IRenderer } from "fela";
import { createRenderer, createSnapshot, cssReboot } from "../../test-helpers";
import { Wrapper as CarouselWrapper } from "./Wrapper";

describe("components/Carousel/Wrapper", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a Carousel Wrapper", async () => {
    const component = <CarouselWrapper>Carousel</CarouselWrapper>;
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
