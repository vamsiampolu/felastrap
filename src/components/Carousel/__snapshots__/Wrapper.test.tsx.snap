// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Carousel/Wrapper should render a Carousel Wrapper 1`] = `
".a {
  position: relative;
}


<div data-testid="carousel" class="a carousel">Carousel</div>
;
"
`;
